src/app/notifications/Notification.tsx

```diff
@@ -47,6 +48,15 @@ export default class Notification extends React.Component<Props, State> {
             <div className="notification-details">
               <div className="notification-title">{notification.title}</div>
               {notification.body && <div className="notification-body">{notification.body}</div>}
+              {needsButton(notification.type) &&
+                notification.buttonEffect &&
+                notification.account && (
+                  <NotificationButton
+                    type={notification.type}
+                    account={notification.account}
+                    effect={notification.buttonEffect}
+                  />
+                )}
```

Comment:
What you're doing here looks like it works great! I did have something a bit different in mind when I made this Notification component, though.

Since `notification.body` can take any React element as an argument, I had figured that adding buttons and other complex content could be as easy as just passing a full React element in as content. That element could have (by closure) a reference to any data or functions it needs (such as the function to undo an action).